plugins/org.komodo.core/src/org/komodo/repository/PropertyDescriptorImpl.java

```diff
@@ -209,4 +215,42 @@ public boolean isMultiple() {
         return this.multiple;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.komodo.spi.repository.PropertyDescriptor#isValid(java.lang.Object)
+     */
+    @Override
+    public boolean isValid( final Object proposedValue ) {
+        if ( ( this.constraints != null ) && ( this.constraints.length != 0 ) ) {
+            if ( Type.STRING == this.type ) {
+                String value = null;
+
+                if ( proposedValue instanceof String ) {
+                    value = ( String )proposedValue;
+                } else if ( proposedValue instanceof Value ) {
+                    try {
+                        value = ( ( Value )proposedValue ).getString();
+                    } catch ( final Exception e ) {
+                        return false;
+                    }
+                } else {
+                    value = proposedValue.toString();
+                }
+
+                for ( final String pattern : this.constraints ) {
+                    if ( value.matches( pattern ) ) {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            // TODO implement other type validation
```

Comment:
For a property with one or more value constraints defined in a CND, the purpose of the `isValid` method is to verify the proposed value passes at least one of those property value constraints. When an invalid value is set on a property that has value constraints defined, an exception is thrown by the JCR framework (ModeShape). I added this method so that the shell could do some validation of property values before letting the JCR framework have its chance.

Here is a link that outlines the JCR value constraint logic that would be needed to be implemented by this method:

http://www.day.com/specs/jcr/1.0/6.7.16_Value_Constraints.html

So we can go about this one of 3 ways:

1. Remove the method altogether and let JCR throw an exception if an invalid value is entered by the user in the shell. Luckily there really aren't that many properties with value constraints.
2. Leave the method in and write an issue up for the remaining work.
3. Finish implementing the method.

I'm leaning toward #1. Wdyt?